spec/editors_draft.html

```diff
@@ -3298,6 +3298,231 @@ <h2>Evaluation Semantics</h2>
+         class="nohighlight example">
+      <!--
+      _:b004 <http://example.org/p3> "[ _:b004 ]"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/List> .
+      -->
+      </pre>
+
+      <p>Loading the Turtle in <a href="#load-renaming-blank-reuse"></a> with a triple with a <a>cdt:Map literal</a> containing two map entries, both having the same blank node value, MUST result in a dataset with a single triple whose <a>cdt:Map literal</a> object contains two map entries having the same blank node value, and whose identifier MUST NOT be "b0" (<a href="#load-renaming-blank-reuse-result"></a>):</p>
+      <pre id="load-renaming-blank-reuse"
+         title="Producing consistent blank node identifiers used multiple times in Composite Values"
+         data-transform="updateExample"
+         data-content-type="text/turtle"
+         class="nohighlight example">
+      <!--
+      ex:s ex:p3 "{ 'this': _:b0, 'again': _:b0 }"^^cdt:Map .
+      -->
+      </pre>
+      <pre id="load-renaming-blank-reuse-result"
+         title="Producing consistent blank node identifiers used multiple times in Composite Values: Result"
+         data-transform="updateExample"
+         data-content-type="text/n-triples"
+         class="nohighlight example">
+      <!--
+      <http://example.org/s> <http://example.org/p3> "{'this': _:b005,'again': _:b005}"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/Map> .
+      -->
+      </pre>
+    </section>
+
+    <section>
+      <h2>Exporting Requirements</h2>
+      <p>
+        Conforming implementations MUST process <a>cdt:List literal</a>s and <a>cdt:Map literal</a>s during export, replacing the substrings containing the serialized form of any blank nodes with identifier <var>id</var> with a blank node label <var>b</var>, such that
+      </p>
+
+      <ul>
+        <li>within the scope of the document being serialized, all occurrences of <var>id</var> as a blank node identifier are replaced with the same <var>b</var> value (both inside and outside of composite values), and</li>
+        <li>within the scope of the document being serialized, two distinct <var>id</var> values are replaced with distinct <var>b</var> values.</li>
+      </ul>
+
+      <p>
+        Here, "exporting" is any single process by which composite values leave a SPARQL system or RDF Dataset. This includes (but is not limited to):
+      </p>
+
+      <ul>
+        <li>Serializing SPARQL Results in a concrete syntax (such as [[SPARQL11-RESULTS-JSON]] or [[SPARQL11-RESULTS-CSV-TSV]])</li>
+        <li>Serializing RDF results to a CONSTRUCT or DESCRIBE query in a concrete syntax (such as [[TURTLE]])</li>
+        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax (such as [[RDF-12-N-QUADS]]) via [[SPARQL12-GRAPH-STORE-PROTOCOL]]</li>
+        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax via means other than [[SPARQL12-GRAPH-STORE-PROTOCOL]]</li>
```

Comment:
```suggestion
        <li>Serializing the result of a SPARQL SELECT or ASK query in a concrete syntax such as [[SPARQL11-RESULTS-JSON]] or [[SPARQL11-RESULTS-CSV-TSV]]</li>
        <li>Serializing the RDF graph resulting from a SPARQL CONSTRUCT or DESCRIBE query in a concrete RDF serialization format such as [[TURTLE]]</li>
        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax (such as [[N-QUADS]]) in response to a SPARQL graph store protocol request [[SPARQL11-HTTP-RDF-UPDATE]]</li>
        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax via means other than responses to SPARQL graph store protocol requests</li>
```

Reply:
I don't think "ASK" queries are relevant here, since their results can't contain anything but a boolean. Otherwise, looks good and will apply these suggestions.

Reply:
You are right, ASK queries are irrelevant here.

Here is the edit suggestion without ASK queries:

```suggestion
        <li>Serializing the result of a SPARQL SELECT query in a concrete syntax such as [[SPARQL11-RESULTS-JSON]] or [[SPARQL11-RESULTS-CSV-TSV]]</li>
        <li>Serializing the RDF graph resulting from a SPARQL CONSTRUCT or DESCRIBE query in a concrete RDF serialization format such as [[TURTLE]]</li>
        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax (such as [[N-QUADS]]) in response to a SPARQL graph store protocol request [[SPARQL11-HTTP-RDF-UPDATE]]</li>
        <li>Serializing the full or partial contents of an RDF dataset in a concrete syntax via means other than responses to SPARQL graph store protocol requests</li>
```